### ISSUE TITLE
crash handling regression in 2.5.6

### ISSUE THREAD

**Issue by cushon** [typetools/checker-framework#2188]
I'm seeing the following crash when upgrading from version 2.5.5 to 2.5.6:

```
warning: The following options were not recognized by any processor: '[printErrorStack]'
error: AsSuperVisitor: type is not an erased subtype of supertype.
  type: A<B extends C, D extends C>
  superType: E<capture#2123 of ? extends C, capture#2124 of ? extends C>; The Checker Framework crashed.  Please report the crash.
error: AsSuperVisitor: type is not an erased subtype of supertype.
  type: @Initialized @NonNull A<B extends @Initialized @NonNull C, D extends @Initialized @NonNull C>  superType: @Initialized @NonNull E<capture#2123 of ? extends @Initialized @NonNull C, capture#2124 of ? extends @Initialized @NonNull C>; The Checker Framework crashed.  Please report the crash.
2 errors
```

The message doesn't include a track trace. Could that be related to af4f59416bc06a5b043d5aedf4ed13f59ee59bca?

Also, the source position of the code that triggered the crash isn't being reported (see #1866).

**Comment by wmdietl** [comment]
https://github.com/typetools/checker-framework/commit/af4f59416bc06a5b043d5aedf4ed13f59ee59bca switched the default to always show the stack trace. The new option `AnoPrintErrorStack` can be used to turn off stack traces.
That explains the warning, but not why you don't get a stack trace.
Maybe the removal of `ErrorReporter` in August https://github.com/typetools/checker-framework/commit/aceb62b0755beada3529e8077d467a9f7102a7c9 caused this and we didn't notice until now?
E.g the logic at https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java#L752
will not show the source position if `ce.getCause() == null`.

`AsSuperVisitor` was touched for the 2.5.6 release: https://github.com/typetools/checker-framework/commit/98801bf7d01d13aac07cdc6d29b9884bb07ec06b
However, that change seems unrelated to the error.
Do you have a minimized test case or did you anonymize real code?

**Comment by cushon** [comment]
The error message was anonymized real code. I don't have a minimal repro yet. It would be easier to produce one if #1866 worked, so I was hoping the cause of the crash handling regression (if not the cause of the crash) might be obvious.

**Comment by wmdietl** [comment]
https://github.com/typetools/checker-framework/pull/2189 should fix the error reporting part.
Could you have a look?

**Comment by cushon** [comment]
> Could you have a look?

Looks good, thanks. I used that to minimize one of the crashes I found: https://github.com/typetools/checker-framework/issues/2190.

There were two others, but I'm hoping they might share a root cause.
